topo_descriptors/topo.py

```diff
@@ -128,10 +132,109 @@ def _tpi_kernel(size):
+        If a scalar, the same fraction is used to determine the smoothing scale
+        of all specified scales. If a list, must match the length of arg 'scales'.
+    ind_nans (optional) : tuple of two 1D arrays
+        Contains the (row, column) indices of the NaNs in the original DEM to be
+        reassigned after computations. NaNs in the original DEM should be
+        interpolated prior computations as they propagate in convolutions with
+        the fast Fourier transform method (scipy.signal.convolve).
+    crop (optional) : dict
+        If specified the outputs are cropped to the given extend. Keys should be
+        the coordinates labels of dem_da and values should be slices of [min,max]
+        extend. Default is None.
+    outdir (optional) : string
+        The path to the output directory. Save to working directory by default.
+
+    See also
+    --------
+    std, circular_kernel
+    """
+
+    hlp.check_dem(dem_da)
+    logger.info(f"***Starting STD computation for scales {scales} meters***")
+    if not hasattr(scales, "__iter__"):
+        scales = [scales]
+    if not hasattr(smth_factors, "__iter__"):
+        smth_factors = [smth_factors] * len(scales)
+
+    scales_pxl, _ = hlp.scale_to_pixel(scales, dem_da)
+    sigmas = hlp.get_sigmas(smth_factors, scales_pxl)
+
+    for idx, scale_pxl in enumerate(scales_pxl):
+        logger.info(
+            f"Computing scale {scales[idx]} meters with smoothing factor"
+            f" {smth_factors[idx]} ..."
+        )
+        name = _std_name(scales[idx], smth_factors[idx])
+        array = std(dem=dem_da.values, size=scale_pxl, sigma=sigmas[idx])
+
+        array[ind_nans] = np.nan
+        hlp.to_netcdf(array, dem_da.coords, name, crop, outdir)
+        del array
+
+
+@hlp.timer
+def std(dem, size, sigma=None):
+    """Compute the standard deviation over a digital elevation model within
+    a rolling window.
+
+    Parameters
+    ----------
+    dem : array representing the DEM.
+    size : int
+        Size of the kernel for the convolution. Represents the diameter (i.e. scale)
+        in pixels at which the std is computed.
+    sigma (optional) : scalar
+        If provided, the DEM is first smoothed with a gaussian filter of standard
+        deviation sigma (in pixel size).
+
+    Returns
+    -------
+    array with local standard deviation values
+
+    See also
+    --------
+    scipy.signal.convolve, scipy.ndimage.gaussian_filter
+    """
+    kernel = circular_kernel(size)
+    kernel_sum = np.sum(kernel)
+    if sigma:
+        dem = ndimage.gaussian_filter(dem, sigma)
+
+    squared_dem = dem.astype("int32") ** 2
+    sum_dem = signal.convolve(dem, kernel, mode="same")
+    sum_squared_dem = signal.convolve(squared_dem, kernel, mode="same")
+
+    variance = (sum_squared_dem - sum_dem ** 2 / kernel_sum) / (kernel_sum - 1)
+    variance = np.clip(variance, 0, None)  # avoid small negative values
+
```

Comment:
That would be my implementation from orginal formula. Is it the same exactly?

local_mean = signal.convolve(dem, kernel / kernel_sum, mode="same") 
local_var = signal.convolve((dem - local_mean)**2, kernel / (kernel_sum -1), mode="same")

return np.sqrt(local_var)

Reply:
mmmh no, it's not the same result. I think that the issue in your implementation is that `(dem - local_mean) ** 2` is not really what you would do in the traditional formula for Var(X) = E[(X - mu)^2], in that mu is not constant in the local window used for the second convolution, right?